.github/workflows/deps.yml

```diff
@@ -0,0 +1,92 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+name: Dependency
+
+on:
+  pull_request:
+    branches:
+      - main
+      - branch-*
```

Comment:
the existing release branch does not has the scripts

Reply:
since the script `dev/dependencies.sh` was introduced in version 0.4.0-SNAPSHOT, which is the current main branch, we don't backport this pull request to an existing release branch, such as branch-0.3. WDYT?

Reply:
we should not backport.

if we merge it as-is to main only, will PR target to branch-0.3 trigger the check?

Reply:
PR target to branch-0.3 will not trigger this check, it's the same as SBT CI.